<a href="https://colab.research.google.com/github/JulianEduardoMendez/programacion1/blob/main/PARCIAL2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nombre: Julian Eduardo Méndez Poveda // Código: 2260668 // PARCIAL II

PARTE 2------------------------

INSTRUCCIÓN 1-----------------------------------------

In [21]:
import numpy as np
import pandas as pd

In [2]:
np.random.seed(0)

In [155]:
datos = np.random.randint(149, 197, 50)
datos

array([169, 177, 168, 151, 175, 169, 189, 181, 178, 196, 187, 189, 157,
       196, 186, 171, 171, 149, 169, 180, 193, 174, 181, 157, 165, 154,
       194, 155, 155, 185, 153, 181, 158, 165, 188, 176, 195, 194, 170,
       169, 195, 180, 180, 184, 179, 154, 169, 183, 162, 171])

In [156]:
filtrado = datos[datos>170]
filtrado

array([177, 175, 189, 181, 178, 196, 187, 189, 196, 186, 171, 171, 180,
       193, 174, 181, 194, 185, 181, 188, 176, 195, 194, 195, 180, 180,
       184, 179, 183, 171])

In [158]:
import pandas as pd
datos = pd.DataFrame(datos)
normalizar = lambda datos: (datos - datos.mean()) / datos.std()
resultado = normalizar(datos)
resultado

,0
0,-0.408133
1,0.181229
2,-0.481803
3,-1.734198
4,0.033888
5,-0.408133
6,1.065272
7,0.475910
8,0.254899
9,1.580964


In [159]:
datos.describe()

,0
count,50.000
mean,174.540
std,13.574
min,149.000
25%,165.750
50%,175.500
75%,184.750
max,196.000


La media de los datos es de 174.54

La desviación estandar es de 13.574 Aproximadamente



El dato mínimo es 149 y el máximo es 196

La cuarta parte de los datos están por debajo de 165.75
La mitad de los datos están por debajo de 175.5
Un 75% de los datos están por debajo de 184.75

In [183]:
datos1 = pd.DataFrame(datos)

In [194]:
datos1

,0
0,169
1,177
2,168
3,151
4,175
5,169
6,189
7,181
8,178
9,196


INSTRUCCIÓN 2--------------------------------------

In [22]:
import pandas as pd

In [185]:
df = pd.DataFrame({
    "Nombre": ["Julián", "Eduardo", "Cesar", "Lucas", "Luis"],
    "Edad": [17, 18, 26, 24, 22],
    "Ciudad": ["Bucaramanga", "Girón", "Lebrija", "Piedecuesta", "Floridablanca"],
    "Dinero": [190, 200, 300, 50, 75],
})

In [186]:
df.head()

,Nombre,Edad,Ciudad,Dinero
0,Julián,17,Bucaramanga,190
1,Eduardo,18,Girón,200
2,Cesar,26,Lebrija,300
3,Lucas,24,Piedecuesta,50
4,Luis,22,Floridablanca,75


In [187]:
df["Nombre"]

,Nombre
0,Julián
1,Eduardo
2,Cesar
3,Lucas
4,Luis


In [188]:
df.loc[df["Edad"] > 20]

,Nombre,Edad,Ciudad,Dinero
2,Cesar,26,Lebrija,300
3,Lucas,24,Piedecuesta,50
4,Luis,22,Floridablanca,75


In [189]:
nuevo = pd.DataFrame({"NuevaColumna": df["Edad"].apply(lambda x: x + 1)
})
nuevo

,NuevaColumna
0,18
1,19
2,27
3,25
4,23


In [190]:
df=df.join(nuevo)

In [191]:
df

,Nombre,Edad,Ciudad,Dinero,NuevaColumna
0,Julián,17,Bucaramanga,190,18
1,Eduardo,18,Girón,200,19
2,Cesar,26,Lebrija,300,27
3,Lucas,24,Piedecuesta,50,25
4,Luis,22,Floridablanca,75,23


In [192]:
grupo = df.groupby("Ciudad")["Dinero"].mean()

In [193]:
grupo

,Dinero
Ciudad,
Bucaramanga,190.0
Floridablanca,75.0
Girón,200.0
Lebrija,300.0
Piedecuesta,50.0


In [174]:
from google.colab import files
import csv
with open("datos_procesados.csv", "w", newline="") as f:
  writer = csv.writer(f)
  writer.writerows(df)

files.download("datos_procesados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

PARTE 3-------------------------------------

Proyecto:

In [48]:
PruebaA = np.random.normal(loc = 7.2, scale = 0.4, size = 53)
PruebaB = np.random.gamma(shape = 306, scale = 0.0229, size = 53)

In [49]:
df0 = pd.DataFrame({
    "A": PruebaA,
    "B": PruebaB
})

In [50]:
df0.describe()

,A,B
count,53.000000,53.000000
mean,7.203057,6.917539
std,0.433997,0.350054
min,6.389982,5.988854
25%,6.905913,6.680107
50%,7.213657,6.951828
75%,7.499286,7.135909
max,8.239322,7.635017


In [51]:
df0.median()

,0
A,7.213657
B,6.951828


La media del A es 7.2 y del B es 6.91 aprox

La desviacion estandar de A es 0.43 y de B es 0.35 aprox

Un cuarto de los datos son inferiores a 6.9 en A y a 6.68 en B

Un 75% de los datos son inferiores a 7.49 en A y a 7.13 en B

El valor máximo de A es 8.23 y de B es 7.63 aprox

El valor mínimo de A es 6.38 y de B es 5.98

La mediana del A es 7.21 y del B es 6.95

In [43]:
from scipy import stats

In [54]:
t_stat, p_val = stats.ttest_ind(df0["A"], df0["B"])

print("Estadístico t:", t_stat)
print("Valor p:", p_val)

Estadístico t: 3.7279171423678408
Valor p: 0.0003144752939359647


No se cumple el supuesto de normalidad con una confianza del 95%, pues el valor p es inferiror a 0.05

In [56]:
stats.ttest_rel(df0["A"], df0["B"])

TtestResult(statistic=np.float64(3.456576363608499), pvalue=np.float64(0.0010984587113451628), df=np.int64(52))

Tambien se rechaza que el puntaje medio en ambas encuestas sea igual con una confianza del 95%, pues el valor p es menor a 0.05

In [60]:
stats.kendalltau(df0["A"], df0["B"])

SignificanceResult(statistic=np.float64(-0.0362844702467344), pvalue=np.float64(0.7013230204757065))

No se rechaza que los puntajes de ambas encuestas no están correlacionado con una confianza del 95%, pues el valor p es mayor a 0.05

In [62]:
stats.spearmanr(df0["A"], df0["B"])

SignificanceResult(statistic=np.float64(-0.06144170295113691), pvalue=np.float64(0.6620753443760858))

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no estan correlacionados

In [63]:
import os
os.makedirs("comparacion", exist_ok=True)

In [78]:
df11 = pd.DataFrame({
    "Estudiante": range(1, 11),
    "Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
    "Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})

In [74]:
%%writefile comparacion/pruebat.py
from scipy import stats
import pandas as pd
df11 = pd.DataFrame({
    "Estudiante": range(1, 11),
    "Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
    "Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})
stats.ttest_rel(df11["Prueba_A"], df11["Prueba_B"])


Overwriting comparacion/pruebat.py


In [79]:
stats.ttest_rel(df11["Prueba_A"], df11["Prueba_B"])

TtestResult(statistic=np.float64(2.282657730758047), pvalue=np.float64(0.04835186429465345), df=np.int64(9))

Bajo la prueba T, se rechaza con una confianza del 95% que el puntaje medio de ambos datos sean iguales

In [81]:
%%writefile comparacion/cor.py
import pandas as pd
from scipy import stats
df11 = pd.DataFrame({
    "Estudiante": range(1, 11),
    "Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
    "Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})
stats.kendalltau(df11["Prueba_A"], df11["Prueba_B"])
stats.spearmanr(df11["Prueba_A"], df11["Prueba_B"])

Overwriting comparacion/cor.py


In [85]:
stats.spearmanr(df11["Prueba_A"], df11["Prueba_B"])

SignificanceResult(statistic=np.float64(0.8395701571521512), pvalue=np.float64(0.002376639566080667))

In [84]:
stats.kendalltau(df11["Prueba_A"], df11["Prueba_B"])

SignificanceResult(statistic=np.float64(0.7063224140220168), pvalue=np.float64(0.006046856693870933))

Bajo las dos pruebas, de correlacion, se rechaza con una confianza del 95% que los puntajes de ambas no están relacionados